packages/patternfly-4/react-charts/src/components/ChartDonut/ChartDonut.tsx

```diff
@@ -338,47 +471,167 @@ export interface ChartDonutProps extends ChartPieProps {
   subTitle,
+  subTitleComponent = <ChartLabel />,
+  subTitleDx = 0,
+  subTitleDy = 0,
+  subTitlePosition = DonutStyles.subTitleOrientation as ChartDonutSubTitlePosition,
   themeColor,
   themeVariant,
   title,
-  width = DonutTheme.pie.width,
+  titleComponent = <ChartLabel />,
 
   // destructure last
-  innerRadius = ((height || width) - 34) / 2,
   theme = getDonutTheme(themeColor, themeVariant),
+  capHeight = title && subTitle ? 1.1 : undefined,
+  height = theme.pie.height,
+  width = theme.pie.width,
+  donutHeight = Math.min(height, width),
+  donutWidth = Math.min(height, width, donutHeight),
+  innerRadius = (Math.min(donutHeight, donutWidth) - 34) / 2,
   ...rest
 }: ChartDonutProps) => {
+  // Returns legend
+  const getLegend = () => {
+    if (!legendData && !legendComponent.props.data) {
+      return null;
+    }
+    const props = legendComponent.props ? legendComponent.props : {};
+    const legendDimensions = getLegendDimensions({
+      legendData,
+      legendOrientation,
+      legendProps: props,
+      theme
+    });
+    return React.cloneElement(legendComponent, {
+      data: legendData,
+      orientation: legendOrientation,
+      standalone: false,
+      theme: theme,
+      x: getLegendX({
+        chartWidth: donutWidth,
+        dx: legendDx,
+        legendData,
+        legendOrientation: props.legendOrientation ? props.legendOrientation : legendOrientation,
+        legendPosition,
+        legendWidth: legendDimensions.width,
+        theme,
+        svgWidth: width
+      }),
+      y: getLegendY({
+        chartHeight: donutHeight,
+        chartType: 'pie',
+        dy: legendDy,
+        legendData: props.data ? props.data : legendData,
+        legendHeight: legendDimensions.height,
+        legendPosition,
+        theme
+      }),
+      ...props
+    });
+  };
+
+  // Returns subtitle
+  const getSubTitle = () => {
+    if (!subTitle || subTitlePosition === ChartDonutSubTitlePosition.center) {
+      return null;
+    }
+    const props = titleComponent.props ? titleComponent.props : {};
+    return React.cloneElement(titleComponent, {
+      style: DonutStyles.label.subTitle,
+      text: subTitle,
+      textAnchor: subTitlePosition === 'right' ? 'start' : 'middle',
+      verticalAnchor: 'middle',
+      x: getLabelX({
+        chartWidth: donutWidth,
+        dx: subTitleDx,
+        labelPosition: subTitlePosition,
+        legendPosition,
+        svgWidth: width
+      }),
+      y: getLabelY({
+        chartHeight: donutHeight,
+        dy: subTitleDy,
+        labelPosition: subTitlePosition
+      }),
+      ...props
+    });
+  };
+
+  // Returns title
+  const getTitle = () => {
+    if (!title) {
+      return null;
+    }
+    const props = titleComponent ? titleComponent.props : {};
+    const showBoth = title && subTitle && subTitlePosition == ChartDonutSubTitlePosition.center;
+    return React.cloneElement(titleComponent, {
+      capHeight: showBoth ? capHeight : undefined,
```

Comment:
Just discovered you can do this to conditionally define a prop:
```suggestion
     ...showBoth && { capHeight }
```

If `showBoth` is false, the capHeight prop won't be set at all, instead of getting `{capHeight: undefined}`

Reply:
Neat. We only need capHeight when both are present. Don't want it applied when there is no subtitle.